### PR TITLE
do client instantiation in _main not top

### PR DESCRIPTION
**Pull request by brantleyr**

### CODE REVIEW

**Comment by kylape** [comment]
Have we considered just setting `maxBytes` on the handler so that it rolls over automatically at a given size?  That would sidestep having to call `doRollover` altogether.

**Comment by brantleyr** [comment]
I don't think its very intuitive to have logs for different execution runtimes dispersed across multiple log files.  Especially in the case of collecting a log file from a customer.

**Comment by gravitypriest** [comment]
Our logs have traditionally been one for each the last three runs.  The current behavior is consistent with that.

**Comment by kylape** [comment]
> I don't think its very intuitive to have logs for different execution runtimes dispersed across multiple log files.

I can see that, particularly with our client typically running once a day.  One log per day.  The only time I could see this being annoying would be if a user was running the client over and over interactively, but this is likely to be quite rare except maybe during development.

> Our logs have traditionally been one for each the last three runs

Really?  I thought they rotated immediately after a successful upload.  This enabled the archive to include all the logs since the last successful upload.

**Comment by gravitypriest** [comment]
Yeah, they rotate after a successful upload. Which means each log is of a single run of the client, and we keep 3 in total.

File: insights_client/__init__.py [hunk 1]
```diff
@@ -123,6 +123,23 @@ def _main():
     attempt to collect and upload with new, then current, then rpm
     if an egg fails a phase never try it again
     """
+     # flake8 complains because these imports aren't at the top
+    from insights.client import InsightsClient  # noqa E402
+    from insights.client import config  # noqa E402
+
+     # handle client instantation here so that it isn't done multiple times in __init__
+    global client, config, debug
```
Comment (kylape) [inline insights_client/__init__.py]: What if we passed the client into the call to phase?  The benefit would be that we wouldn't need these global vars.  wdyt?

**Comment by kylape** [comment]
> we keep 3 in total.

Ah gotcha.

File: insights_client/__init__.py [hunk 2]
```diff
@@ -123,6 +123,23 @@ def _main():
...
+    config = client.get_conf()
+    debug = config["debug"]
+
+    # handle log rotation here instead of core
+    if os.path.isfile(config['logging_file']):
+        log_handler = logging.handlers.RotatingFileHandler(
+            config['logging_file'], delay=True, backupCount=3)
+        log_handler.doRollover()
+    # we now have access to the clients logging mechanism instead of using print
+    client.set_up_logging()
```
[the remark this answers is not in the document]
Comment (brantleyr) [reply insights_client/__init__.py]: We can keep print for now I guess. Should we need access to the same logging handler in the future then its there.

**Pull request merged**